### PR TITLE
ledger-on-sql: Switch to the extra-simplified KVUtils API.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
I think this will make it much easier to implement static time for #4499.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/queries/CommonQueries.scala [hunk 1]
```diff
@@ -32,38 +30,28 @@ trait CommonQueries extends Queries {
...
           case index ~ entryId ~ envelope =>
-            index -> LedgerRecord(
-              Offset(Array(index)),
-              DamlLogEntryId
-                .newBuilder()
-                .setEntryId(ByteString.copyFrom(entryId))
-                .build(),
-              envelope,
-            )
+            index -> LedgerRecord(Offset(Array(index)), DamlLogEntryId.parseFrom(entryId), envelope)
```
Comment (miklos-da) [inline ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/queries/CommonQueries.scala]: I've made a note to change `LedgerRecord` to hide serialization aspects (i.e., to just ask for the bytes that have been written via `appendToLog` so you won't have to take care of parsing the log entry's key).

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LedgerStateAccess.scala [hunk 2]
```diff
@@ -55,25 +55,30 @@ trait LedgerStateOperations {
   */
 abstract class BatchingLedgerStateOperations(implicit executionContext: ExecutionContext)
     extends LedgerStateOperations {
-  override def readState(key: Key): Future[Option[Value]] =
+  override final def readState(key: Key): Future[Option[Value]] =
```
Comment (miklos-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LedgerStateAccess.scala]: Thanks for these improvements!

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala [hunk 3]
```diff
@@ -131,20 +131,20 @@ final class ApiSubmissionService private (
                 Success(())
 
               case Success(Overloaded) =>
-                logger.debug(s"Submission has failed due to back pressure")
+                logger.info(s"Submission has failed due to back pressure")
```
Comment (miklos-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala]: Unnecessary string interpolation.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala [hunk 4]
```diff
@@ -131,20 +131,20 @@ final class ApiSubmissionService private (
...
 
               case Success(NotSupported) =>
-                logger.debug(s"Submission of command was not supported")
+                logger.warn(s"Submission of command was not supported")
                 Failure(Status.INVALID_ARGUMENT.asRuntimeException)
 
               case Success(InternalError(reason)) =>
-                logger.debug(
+                logger.error(
                   s"Submission of command failed due to an internal error, reason=$reason ")
```
Comment (miklos-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala]: Unnecessary space at the end of the string.

File: ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala [hunk 5]
```diff
@@ -58,68 +65,69 @@ class SqlLedgerReaderWriter(
...
+                }
+              })
+            .mapMaterializedValue(_ => NotUsed)
         }),
       )
       .map { case (_, record) => record }
 
   override def commit(correlationId: String, envelope: Array[Byte]): Future[SubmissionResult] =
     withEnrichedLoggingContext("correlationId" -> correlationId) { implicit logCtx =>
-      Future {
```
Comment (miklos-da) [inline ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala]: Nice negative delta.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala]: Took care of this, and others around it. Thanks!

**Pull request merged**